daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SResult.scala

```diff
@@ -18,7 +18,9 @@ sealed abstract class SResult extends Product with Serializable
 
 object SResult {
   final case class SResultError(err: SError) extends SResult
-  final case object SResultContinue extends SResult
+
+  /** The speedy machine has completed evaluation to reach a final value.  */
+  final case class SResultFinalValue(v: SValue) extends SResult
```

Comment:
Having `SResultFinalValue` carry the value as payload avoids having the caller dig around for it in the `Ctrl` of the machine, and knowing internal details of the machine's operation.

I think most callers dont really care about the final value because they are running the machine only for it's effect, or for other components, i.e. `Engine.scala` cares about `machine.ptx`.

There is one caller that makes use the payload:
 daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

Reply:
That makes sense. Thanks.